homeassistant/components/light/mochad.py

```diff
@@ -0,0 +1,94 @@
+"""
+Contains functionality to use a X10 dimmer over Mochad.
+
+For more details about this platform, please refer to the documentation at
+https://home.assistant.io/components/light.mochad/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, SUPPORT_BRIGHTNESS, Light, PLATFORM_SCHEMA)
+from homeassistant.components import mochad
+from homeassistant.const import (CONF_NAME, CONF_PLATFORM, CONF_DEVICES,
+                                 CONF_ADDRESS)
+from homeassistant.helpers import config_validation as cv
+
+DEPENDENCIES = ['mochad']
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PLATFORM): mochad.DOMAIN,
+    CONF_DEVICES: [{
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Required(CONF_ADDRESS): cv.x10_address,
+        vol.Optional(mochad.CONF_COMM_TYPE): cv.string,
+    }]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up X10 dimmers over a mochad controller."""
+    devs = config.get(CONF_DEVICES)
+    add_devices([MochadLight(
+        hass, mochad.CONTROLLER.ctrl, dev) for dev in devs])
+    return True
+
+
+class MochadLight(Light):
```

Comment:
Please see the earlier comments in this PR on this. We don't want to rely on the mochad daemon to track the device status because the x10 devices are 1 way and it doesn't actually know the state. (it just stores a record of the last command sent) It also causes a synchronization issue because the daemon is a serial interface. Having home assistant track the device state is much more reliable. The only reason I poll for the status on __init__ is to take a better guess at the initial state.

The read_data() call in all the control functions is just for synchronization, it is waiting for the ack from the mochad daemon on the socket interface.